### PR TITLE
feature: add sequential downloader for s3

### PR DESCRIPTION
**Pull request by bryancua**

### CODE REVIEW

File: s3/download.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type sequentialWriterAt struct {
+	w io.Writer
+}
+
+func (sw sequentialWriterAt) WriteAt(p []byte, offset int64) (int, error) {
+	return sw.w.Write(p)
```
Comment (lostlevels) [inline s3/download.go:17]: Won't this cause issues if concurrency is > 1?
Might want to wrap https://docs.aws.amazon.com/sdk-for-go/api/aws/#WriteAtBuffer
if so

Comment (bryancua) [reply s3/download.go:17]: It's meant to be used only when concurrency = 1 .. .so private.

Comment (bryancua) [reply s3/download.go:17]: I didn't really want to store the buffer contents in memory at all (could be gigs).

Comment (lostlevels) [reply s3/download.go:17]: oh i didn't see it was private

Comment (bryancua) [reply s3/download.go:17]: =)

Comment (bryancua) [reply s3/download.go:17]: Thanks for reading it carefully ^__^ much helpful

**Pull request merged**